tracks/rust/exercises/poker/mentoring.md

```diff
@@ -0,0 +1,232 @@
+### Concepts
+
+- Iterators
+- Iterator Methods
+- Pattern matching
+- Modelling a problem with the type system
+
+### Reasonable Solutions
+
+A reasonable solution should:
+
+- Use iterator whenever possible instead of for loops with conditionals inside
```

Comment:
This advice always makes me skeptical. Rust's iterators are powerful and nice, but it's not justified to claim that they should always be preferred over for loops. In particular, if contained code requires a bunch of `?` operators, for loops can be much simpler to use.

Reply:
In general I completely agree with you. However in this case, the emphasis is on the `conditionals inside` rather than having a for loop with some ifs and complicated logic, having iterators with combinators (.filter and the like) usually makes intent much clearer (IMO)